jina/executors/dump.py

```diff
@@ -0,0 +1,147 @@
+import os
+import sys
+from typing import Iterable, Tuple, List
+
+import numpy as np
+
+from jina.enums import BetterEnum
+
+BYTE_PADDING = 4
+
+
+class DumpTypes(BetterEnum):
+    """The enum of dump formats"""
+
+    DEFAULT = 0
+
+
+class DumpPersistor:
+    """
+    Class for creating and importing from dumps
+
+    Do NOT instantiate. Only provides static methods
+    """
+
+    @staticmethod
+    def export_dump_streaming(
+        path: str,
+        shards: int,
+        size: int,
+        data: Iterable[Tuple[str, np.array, bytes]],
+        formats: List[DumpTypes],
+    ):
+        """Export the data to a path, based on sharding,
+
+        :param path: path to dump
+        :param shards: the nr of shards this pea is part of
+        :param size: total amount of entries
+        :param data: the generator of the data (ids, vectors, metadata)
+        :param formats: the list of formats in which we dump
+        """
+        for format in formats:
+            if format == DumpTypes.DEFAULT:
+                DumpPersistor._handle_dump(data, path, shards, size)
+            else:
+                raise NotImplementedError('Not other format types supported right now')
+
+    @staticmethod
+    def _handle_dump(data, path, shards, size):
+        if os.path.exists(path):
+            raise Exception(f'path for dump {path} already exists. Not dumping...')
+        size_per_shard = size // shards
+        extra = size % shards
+        shard_range = list(range(shards))
+        for shard_id in shard_range:
+            if shard_id == shard_range[-1]:
+                size_this_shard = size_per_shard + extra
+            else:
+                size_this_shard = size_per_shard
+            DumpPersistor._write_shard_data(data, path, shard_id, size_this_shard)
+
+    @staticmethod
+    def _write_shard_data(data, path, shard_id, size_this_shard):
+        shard_path = os.path.join(path, str(shard_id))
+        shard_docs_written = 0
+        os.makedirs(shard_path)
+        vectors_fp, metas_fp, ids_fp = DumpPersistor._get_file_paths(shard_path)
+        with open(vectors_fp, 'wb') as vectors_fh:
```

Comment:
u can use the 3 context managers in one line

Reply:
Done, check again if you can